components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/ThemeManagerHostObject.java

```diff
@@ -0,0 +1,437 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.appmgt.hostobjects;
+
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jaggeryjs.hostobjects.file.FileHostObject;
+import org.jaggeryjs.scriptengine.exceptions.ScriptException;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.Function;
+import org.mozilla.javascript.NativeArray;
+import org.mozilla.javascript.NativeObject;
+import org.mozilla.javascript.Scriptable;
+import org.mozilla.javascript.ScriptableObject;
+import org.wso2.carbon.appmgt.api.AppManagementException;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.apache.commons.io.FileUtils;
+
+/**
+ * This class has methods to add,remove custom themes.
+ * <p/>
+ * Default theme (store/themes/<themeName>) will be overriden in store/themes/<tenantDomain>/themes/custom.
+ * <p/>
+ * Extension level theme (store/extensions/assets/<assetType>/themes/<themeName>) will be overriden in
+ * store/themes/<tenantDomain>/extensions/assets/<assetType>/themes/custom. Theme type is equal to asset type will be
+ * same  extension level themes.
+ * <p/>
+ * Only whitelisted files will be allowed to uploaded with custom theme. Whitelisted file extension are configured in
+ * admin-dasboard/site/config/site.json
+ */
+public class ThemeManagerHostObject extends ScriptableObject {
+    private static final Log log = LogFactory.getLog(ThemeManagerHostObject.class);
+    private static final String DEFAULT = "default";
+
+    /**
+     * Construct and return the default theme path of store.
+     * [/repository/deployment/server/jaggeryapps/store/themes]
+     *
+     * @return path
+     */
+    private static String getStoreThemePath() {
+        StringBuilder path = new StringBuilder("repository").append(File.separator).append("deployment").append(
```

Comment:
> Check api "path" of java 1.7 -> which will handle file.seperator

See [Paths.get(..)](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Paths.html#get(java.lang.String,%20java.lang.String...)) and [Path.resolve(...)](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#resolve(java.lang.String)) methods